### DISCUSSION TITLE
Woche 2

### DISCUSSION THREAD

**Discussion by TimSchnur** [Christian-2003/CrInGE#4] [Student Exchange]
Hallo zusammen,

herzlich willkommen zu Woche 2. Heute möchten wir euch zuerst unser 👥  **Team** vorstellen. Danach möchten wir die 💡 **Vision** unseres Projektes und die damit verbundenen Plattformen und Techniken vorstellen. Anschließend möchten wir euch den Backlog für den ersten Sprint zeigen.

###### Team
Als erstes möchten wir euch unser 👥 **Team** vorstellen.
| Name | Aufgabe |
|--------|--------|
| @Elekt0 (Nunn, Jannik) | Developer 💻 |
| @FabianDev001 (Nuß, Kai Fabian) | Designer & Developer 🎨 | 
| @TimSchnur (Schnur, Tim) | Scrum Master & Developer 🤺 |
| @Flinks271 (Solfronk, Felix Franz Anton) | Software Architekt & Developer 🏠 |
| @Christian-2003 (Steinbring, Christian) | Product Owner & Developer 🏬 |

So viel zum Team, mit welchem wir unser Projekt realisieren wollen. Da wir schon vom Projekt sprechen, wollt Ihr unsere Vision kennenlernen?

###### Vision

Hat irgendwer von euch schonmal versucht ein 🎲 Videospiel zu entwickeln? Diejenigen von euch, die das schonmal versucht haben werden wissen, dass das mit viel Aufwand verbunden ist.

Daher wollen wir mit _CrInGE_ (<ins>C</ins>ompute<ins>r</ins>ized <ins>In</ins>tegrated <ins>G</ins>ame <ins>E</ins>ngine) euch die Entwicklung von zweidimensionalen Videospielen vereinfachen. Stellt euch vor, ihr könntet Spiele wie "_Geometry Dash_" oder "_Temple Run_" selbst programmieren.

Kurzgesagt ist die 💡 **Vision** hinter _CrInGE_ einen Designer für zweidimensionale Spiele zu programmieren.

Am Ende soll es möglich sein Spiele wie "_Geometry Dash_", "_Tempel Run_" oder "_Subway Surfer_" per Drag and Drop zu entwickeln und exportieren. Hierbei soll man Kollisionen und den Hintergrund der Welt einstellen, sowie dynamische Elemente und Bewegungen hinzufügen können.

Dies wollen wir mit 🍵 **Java** als Desktop Anwendung umsetzen, die mit Maven gebaut werden soll.
Außerdem versuchen wir möglichst auf externe Bibliotheken zu verzichten 💪🏼.

###### Plattformen

Für die Planung benutzen wir [GitHub Projects](https://github.com/users/Christian-2003/projects/2), wo ihr unsere jeweils zweiwöchigen Sprints finden könnt.

Der Fortschritt wird mit unseren regelmäßigen Daily's 💬 2 mal wöchentlich aktualisiert.

Außerdem ist im verlauf der Woche hier auch unser [momentaner Sprint Backlog](https://github.com/users/Christian-2003/projects/2/views/9) und das [momentane Kanban Board](https://github.com/users/Christian-2003/projects/2/views/6) zu finden.

<br>

Hoffentlich konnten wir euch unser Projekt etwas näher bringen. Wir freuen uns darauf, euch nächste Woche erneut unseren Fortschritt zu präsentieren.

**Comment by PetrusQn** [comment]
Hi CrInGE-Team,

Anhand des Designs eures Blogeintrags ist zu vermuten, dass ihr ambitioniert und motiviert seid. Das gefällt mir, da ich mich früher schon gefragt habe, wie die genannten 2D-Spiele funktionieren. 
Daher erhoffe ich mir von eurem Projekt spannende Einblicke, in Form euerer Blogeinträge, um meinen Horizont im Sinne der (2D-)Spieleentwicklung erweitern zu können. Ich bin überzeugt, dass jeder BWL-Student in Zukunft, während einer langweiligen Vorlesung, zum "2D-Spieleentwickler" wird. Irgendwann stellt dieser dann fest, dass er besser Informatik studieren sollte 😆 
Bemerkenswert ist natürlich auch die Vollständigkeit und Übersichtlichkeit des Eintrags, sodass nahezu keine Fragen offen bleiben. Es gibt jedoch eine Ausnahme:
Aus welchem Grund wollt ihr auf externe Bibliotheken verzichten? Unter Benutzung dieser könntet ihr bspw. schneller zum Ziel kommen...

Schlussendlich ist der Eintrag (optisch und inhaltlich) sehr ansprechend und lässt nur wenig Raum für Unklarheiten. Zusätzlich wird eine Erwartungshaltung und Spannung bzgl. des weiteren Verlaufs des Projekts aufgebaut. Weiter so!

**Reply by FabianDev001** [reply]
Zunächst einmal vielen Dank für euer Feedback.

In Bezug auf deine Frage: Unser Ziel bei _CrInGE_ besteht darin, eine möglichst eigenständige Anwendung zu entwickeln. Durch den Verzicht auf externe Bibliotheken versuchen wir das Thema der Spieleentwicklung noch tiefer zu durchdringen. Deshalb wollen wir insbesondere die Game Engine selbst implementieren und so die Verwendung externer Bibliotheken vermeiden.

Da es jedoch schwierig werden könnte gänzlich auf externe Bibliotheken zu verzichten, behalten wir uns vor, nach einer sorgfältigen internen Abwägung der Vor- und Nachteile für das Projekt, trotzdem kleinere externe Bibliotheken einzubinden.

**Comment by lukpr0** [comment]
Ihr habt ein sehr interessantes Thema gewählt: Ich bin sehr gespannt, auf eure Ergebnisse, Jedoch frage ich mich, was für Funktionen ihr selbst implementieren wollt, statt existierende Bibliotheken zu verwenden und warum

**Reply by TimSchnur** [reply]
Wir wollen uns möglichst eine Herausforderung für das Projekt stellen 💪🏼.
Deshalb wollen wir Funktionen wie die Game Physik selbst implementieren und auf bestehende, externe Engines verzichten.
